### PR TITLE
Using the latest metrics library and reporting per-tenant span normal…

### PR DESCRIPTION
**Pull request by buchi-busireddy**
…ization time.

This depends on https://github.com/hypertrace/service-framework/pull/8

### CODE REVIEW

File: span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java [hunk 1]
```diff
@@ -4,19 +4,31 @@
...
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class JaegerSpanNormalizerTest {
+  private final Random random = new Random();
+
+  @BeforeAll
+  static void initializeMetricRegistry() {
```
Comment (tim-mwangi) [inline span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java:27]: Do you need to clear the status of this and isn't it better to do on BeforeEach and clear in AfterEach?

File: span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java [hunk 2]
```diff
@@ -122,19 +134,30 @@ public RawSpan convert(Span jaegerSpan) {
...
-    return rawSpan;
+    // Record the time taken for converting the span, along with the tenant id tag.
+    return tenantToSpanNormalizationTimer
+        .computeIfAbsent(
+            tenantId.get(),
+            tenant ->
+                PlatformMetricsRegistry.registerTimer(
+                    SPAN_NORMALIZATION_TIME_METRIC, Map.of("tenantid", tenant)))
+        .recordCallable(
+            () -> {
```
Comment (tim-mwangi) [inline span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java:145]: Wrap this logic in a method.

File: span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java (hunk 1 above)
Comment (buchi-busireddy) [reply span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java:27]: The test is using the registry similar to a regular service would use it. It's initialized once and each test is like a separate thread/method. Hence, no need to clear the registry for this test.

File: span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java (hunk 2 above)
Comment (buchi-busireddy) [reply span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java]: Done.

File: span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java (hunk 1 above)
Comment (tim-mwangi) [reply span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java:27]: ok.

**Comment by tim-mwangi** [comment]
lgtm but snyk-scan is failing.

**Pull request merged**